crates/astria-composer/src/searcher/rollup.rs

```diff
@@ -0,0 +1,135 @@
+//! Parsing strings of the form `<chain_id>::<url>`
+
+use std::fmt;
+
+use once_cell::sync::Lazy;
+use regex::Regex;
+
+#[derive(Debug)]
+pub(super) struct Rollup {
+    chain_id: String,
```

Comment:
in a later PR `chain_id` should be made into a non-`String` type to enforce that it is a valid chain id format, similar to what is done in [`sequencer-types`](https://github.com/astriaorg/astria/blob/main/crates/astria-sequencer-types/src/namespace.rs#L25).

or maybe just use that type here?

a `chain_id` here gets mapped directly into a namespace, anyways, as far as im aware

Reply:
Yeah, agree. But that's orthogonal to this PR and requires some more thinking around sequencer's public interfaces/API